src/domain/hooks/helpers/event-cache.helper.ts

```diff
@@ -163,6 +147,29 @@ export class EventCacheHelper implements OnModuleInit, OnModuleDestroy {
     await this.cacheService.increment(cacheKey, MAX_TTL);
   }
 
+  /**
+   * Logs the number of unsupported chain events for each chain and clears the store.
+   * This function is public just for testing purposes.
+   */
+  @Interval(EventCacheHelper.UNSUPPORTED_EVENTS_LOG_INTERVAL)
+  public async logUnsupportedEvents(): Promise<void> {
+    await Promise.all(
+      this.unsupportedChains.map(async (chainId) => {
+        const cacheKey = CacheRouter.getUnsupportedChainEventCacheKey(chainId);
+        const count = await this.cacheService.getCounter(cacheKey);
+        if (count) {
+          this.loggingService.warn({
+            type: EventCacheHelper.UNSUPPORTED_CHAIN_EVENT,
+            chainId,
+            count,
+          });
+          await this.cacheService.deleteByKey(cacheKey);
+        }
+      }),
+    );
+    this.unsupportedChains = [];
```

Comment:
There are no changes to the function body, I've changed its position in the file to reflect this is `public` now (following the convention of having public functions before the private ones).